src/rgw/rgw_rest_pubsub.cc

```diff
@@ -423,6 +423,48 @@ auto find_unique_topic(const rgw_pubsub_bucket_topics& bucket_topics, const std:
 }
 }
 
+void remove_notification_by_topic(const DoutPrefixProvider *dpp, const std::string& topic_name, const RGWPubSub::BucketRef& b, optional_yield y) {
+  std::optional<RGWPubSub> ps;
```

Comment:
why not passing RGWPubSub as a parameter as well?

Reply:
I thought that as the definition of `ps` must be present in rgw_pubsub.h so I've just declared it directly. Is this wrong?

Reply:
its not wrong. just unnecessary.
if you have a "ps" local variable in the calling function, just pass it in, instead of creating another one.
this should be similar to how you did it with `RGWPubSub::BucketRef`